src/hu/elte/txtuml/stdlib/util/UUIDImplementation.java

```diff
@@ -0,0 +1,112 @@
+package hu.elte.txtuml.stdlib.util;
+
+/**
+ * Implementation class for UUID interface for txtUML models.
+ * 
+ * @see UUID
+ * @author Salhi Fadhel (fadhelsalhif@gmail.com)
+ *
+ */
+
+public class UUIDImplementation implements UUID {
+
+	/**
+	 * Store UUID object and then do all other functions with it
+	 */
+	private java.util.UUID var;
+
+	/**
+	 * Instance method that returns val,
+	 * 
+	 * @param val
+	 * @return
+	 */
+	protected java.util.UUID getOrigUUID(Object val) {
+		if (val instanceof UUIDImplementation) {
+			return ((UUIDImplementation) val).var;
+		} else if (val instanceof UUID) {
+			return var;
+		} else
+			return null;
+	}
+
+	/**
+	 * Constructs a new UUID using the specified data. mostSigBits is used for
+	 * the most significant 64 bits of the UUID and leastSigBits becomes the
+	 * least significant 64 bits of the UUID.
+	 * 
+	 * @param mostSigBits
+	 *            - The most significant bits of the UUID
+	 * @param leastSigBits
+	 *            - The least significant bits of the UUID
+	 */
+	public UUIDImplementation(long mostSigBits, long leastSigBits) {
+		this.var = new java.util.UUID(mostSigBits, leastSigBits);
+	}
+
+	/**
+	 * protected constructor
+	 * 
+	 * @param var
+	 */
+	protected UUIDImplementation(java.util.UUID var) {
+		this.var = var;
+	}
+
+	@Override
+	public int clockSequence() {
+
+		return var.clockSequence();
+	}
+
+	@Override
+	public int compareTo(UUID val) {
+		return var.compareTo(getOrigUUID(val));
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
The `obj` parameter should be type `UUID` instead of `Object` because `Object` is not a txtUML class nor an external class.